test/e2e/tuning_options_test.go

```diff
@@ -320,3 +327,195 @@ func TestHAProxyTimeoutsRejection(t *testing.T) {
+				fmt.Sprintf("https://%s", canaryRoute.Spec.Host),
+			}
+			if err := podExec(t, *curlPod, &stdout, &stderr, cmd); err != nil {
+				t.Fatalf("Failed running command %q: %v\nstdout:\n%s\nstderr:\n%s", cmd, err, stdout.String(), stderr.String())
+			}
+
+			// Find the router pod for our ingress controller so we can check its logs. This ingress controller specifies
+			// replicas=1, so there will only be one.
+			routerPodList, err := getPods(t, kclient, routerDeployment)
+			if err != nil {
+				t.Fatalf("failed to get pods in deployment %s: %v", routerDeploymentName, err)
+			} else if len(routerPodList.Items) == 0 {
+				t.Fatalf("no pods found in deployment %s", routerDeploymentName)
+			}
+			routerPod := routerPodList.Items[0]
+
+			// Get the logs from the router pod.
+			kubeConfig, err := config.GetConfig()
+			if err != nil {
+				t.Fatalf("failed to get kube config: %v", err)
+			}
+
+			client, err := kubernetes.NewForConfig(kubeConfig)
+			if err != nil {
+				t.Fatalf("failed to create kube client: %v", err)
+			}
+
+			logsContainerName := "logs"
+			readCloser, err := client.CoreV1().Pods(routerPod.Namespace).GetLogs(routerPod.Name, &corev1.PodLogOptions{
+				Container: logsContainerName,
+				Follow:    false,
+			}).Stream(context.TODO())
+			if err != nil {
+				t.Fatalf("Failed to read logs from pod %s container %s: %v", routerPod.Name, logsContainerName, err)
+			}
+
+			defer func() {
+				if err := readCloser.Close(); err != nil {
+					t.Logf("Failed to close reader for pod %s container %s: %v", routerPod.Name, logsContainerName, err)
+				}
+			}()
+
+			// Search the logs container for the specified cookie, making sure that it exactly matches the expected truncated value.
+			expectedCookieString := fmt.Sprintf("%s=%s", cookieName, padding)
+			cookieNameRegexp := regexp.MustCompile(fmt.Sprintf("cookie: (%s=a+x?)$", cookieName))
+			scanner := bufio.NewScanner(readCloser)
+			for scanner.Scan() {
+				line := scanner.Text()
+				// FindSubmatch returns a slice of []byte. Index 0 contains the entire matched string, and each following index
+				// contains one of the submatches (sections of the regexp in parentheses). There's only 1 submatch in this
+				// regexp, so it should either return nil if there's no match, or a 2-item slice.
+				if match := cookieNameRegexp.FindSubmatch([]byte(line)); match != nil {
+					if len(match) != 2 {
+						// It shouldn't be possible to match the regexp without the submatch, but this check will cover our
+						// bases.
+						t.Fatalf("Found regexp match but not submatch")
+					}
+					cookieValue := match[1]
+					if len(cookieValue) == maxLength {
+						return
+					}
+					t.Fatalf("Found cookie but length did not match. Expected %d, got %d", maxLength, len(cookieValue))
+				}
+			}
+			t.Errorf("Cookie %q was not found in router pod %q logs", expectedCookieString, routerPod.Name)
```

Comment:
```suggestion
			// Search the logs container for the specified
			// cookie, making sure that it exactly matches
			// the expected truncated value.
			expectedTruncatedCookieString := fmt.Sprintf("%s=%s", cookieName, padding) // this is without the 'x'
			foundExpectedCookie := false
			scanner := bufio.NewScanner(readCloser)
			for scanner.Scan() {
				line := strings.TrimSpace(scanner.Text())
				if strings.Contains(line, cookieString) {
					t.Fatalf("found untruncated cookie string: %v", cookieString)
				}
				if strings.Contains(line, expectedTruncatedCookieString) {
					foundExpectedCookie = true
				}
			}
			if !foundExpectedCookie {
				t.Fatalf("did not find expected truncated cookie string: %v", expectedTruncatedCookieString)
			}
```

Reply:
I changed the test to also assert that we don't match the cookie that has the trailing "x" in addition to asserting that there's a match for the truncated value.

Reply:
If it's easier to consume then here's my suggestion with the addition that we check the scanner for errors too:

```go
			// Search the logs container for the specified
			// cookie, making sure that it exactly matches
			// the expected truncated value.
			expectedTruncatedCookieString := fmt.Sprintf("%s=%s", cookieName, padding) // this is without the 'x'
			foundExpectedCookie := false
			scanner := bufio.NewScanner(readCloser)
			for scanner.Scan() {
				line := strings.TrimSpace(scanner.Text())
				if strings.Contains(line, cookieString) {
					t.Fatalf("found untruncated cookie string: %v", cookieString)
				}
				if strings.Contains(line, expectedTruncatedCookieString) {
					foundExpectedCookie = true
				}
			}
			if err := scanner.Err(); err != nil {
				t.Fatalf("Error while reading container logs: %v", err)
			}
			if !foundExpectedCookie {
				t.Fatalf("did not find expected truncated cookie string: %v", expectedTruncatedCookieString)
			}
```

Reply:
Thanks, this does make it much more straightforward. I've incorporated it in my latest commit.